core-bundle/contao/dca/tl_user.php

```diff
@@ -582,9 +582,11 @@ public function switchUser($row, $href, $label, $title, $icon)
 	 */
 	public function sessionField(DataContainer $dc)
 	{
+		$allowedOperations = $GLOBALS['TL_DCA']['tl_user']['fields']['session']['eval']['operations'] ?? array();
+
 		if (Input::post('FORM_SUBMIT') == 'tl_user')
 		{
-			$arrPurge = Input::post('purge');
+			$arrPurge = array_values(array_intersect(Input::post('purge'), $allowedOperations));
 
 			if (is_array($arrPurge))
 			{
```

Comment:
The is_array check needs to be moved up I think. Otherwise if no checkbox is checked `Input::post('purge')` could be `null` and cause the `array_intersect()` call to fail.

Or we could use an array cast `array_intersect((array) Input::post('purge'),…`

Reply:
or `Input::post('purge') ?? []` for better readability?

Reply:
Done, moved it down so that we leave as many lines untouched as possible.